Added ability to retrieve all recipes for a product

miapuffia opened the pull request:
I made changes to return all known recipes for a given product. As it is right now, however, only the first recipe is shown as I haven't gotten the UI changes added yet. All of those recipes currently get logged to the MelonLoader console. I don't have many recipes learned in any of my saves so I can't test for performance very well.

I also fixed recipes not being shown when the product is its own ingredient (like Grandaddy Purple = Grandaddy Purple + Flu Medicine)

Please give it some testing while I work on the UI changes. Thanks for the great mod!

robbmanes commented:
Hey, thank you very much for the PR.  I need to shim this into the fix today that fixes the product ordering.  Love this commit though, after work today I'll take a look.

miapuffia commented:
In my latest commit I-

1. Added the multiple recipes to the UI
2. Fixed the very large + icons
3. Added an Expanded Recipe(s) text label to improve the look

When you get to testing it, replace the project file with your own as mine has the game location on a different drive letter.

miapuffia commented:
From what I can see looking at conflicts, my PR also fixed the recipe order issue and can be used in its entirety.

robbmanes commented:
Thanks for this!  I've also worked on the recursion a bit and will do a review and merge of our combined efforts with credit to you, appreciate the work.  I'll get to this ASAP.

robbmanes commented:
@miapuffia I factored in some of what you've done other than the UI changes into this commit (https://github.com/robbmanes/Schedule1_ExpandRecipe/commit/937ba1b40fda88b25247ab5432202d0c18ce8949), and will shim your UI changes in there.  I don't like the idea of having multiple files currently so I'll combine them and fix some of the merge/formatting if you don't mind.

The credit via CoAuthor *should* work, let me know if it's not, and the commit above is very WIP, but I ran out of time today.  In testing I found it extremely performant for recursion, but still need to add the tree flattening to the final lists to return only full recipes.  I'll try to do that tomorrow.

Thanks!

robbmanes commented:
Final recursion patch [here](https://github.com/robbmanes/Schedule1_ExpandRecipe/commit/0cb1eed7306db5595055c522cce25f3da877153e), working on UI next

robbmanes commented:
Shimmed your commits into https://github.com/robbmanes/Schedule1_ExpandRecipe/compare/main...expand_all_recipes - having friends test it thoroughly and then will push a new update.  Thanks again @miapuffia, especially for the UI work, it really helped.

miapuffia commented:
No problem, I'm glad to see the changes in! And I didn't know commit co-author was a feature but it looks like it worked. I'll go ahead and close out this PR. Cheers!
